### ISSUE TITLE
How do I add multiple loss in the configuration file?

### ISSUE THREAD

**Issue by bmxbmx3** [open-mmlab/mmclassification#796]
推荐使用英语模板 General question，以便你的问题帮助更多人。

### 首先确认以下内容
- 我已经查询了相关的 issue，但没有找到需要的帮助。
- 我已经阅读了相关文档，但仍不知道如何解决。

### 描述你遇到的问题

I want to known how to add multiple loss in the configuration file?

### 相关信息

![IEJ{FT39%6~ G(4~5IZ8{6Y](https://user-images.githubusercontent.com/33427015/164682980-4dcf1bef-c50f-4404-9226-426807293762.png)
I want to combine Focal loss and CrossEntropy loss with different loss weight as a new kind of loss in the configuration file，but haven't found the help information related in the documentation，so how to solve this？

**Comment by bmxbmx3** [comment]
![image](https://user-images.githubusercontent.com/33427015/164684968-3105d854-0d00-4453-aca0-0a36a1d93601.png)
so what's the meaning of the 'loss_weight' parameter in this picture?

> Sorry that MMClassification does not support this feature currently.
> 
> You can only create a `MutilLossWarper` class to add this feature.

**Comment by Ezra-Yu** [comment]
Just scale the loss now. here is the [code](https://github.com/open-mmlab/mmclassification/blob/6beac5060f0c1d57141642dba77c19c9ed6b3873/mmcls/models/losses/focal_loss.py#L108) and the [doc](https://github.com/open-mmlab/mmclassification/blob/6beac5060f0c1d57141642dba77c19c9ed6b3873/mmcls/models/losses/cross_entropy_loss.py#L144)

The `loss` field only supports `dict` data, refer to this [line](https://github.com/open-mmlab/mmclassification/blob/6beac5060f0c1d57141642dba77c19c9ed6b3873/mmcls/models/heads/cls_head.py#L32)

**Comment by bmxbmx3** [comment]
> Just scale the loss now. here is the [code](https://github.com/open-mmlab/mmclassification/blob/6beac5060f0c1d57141642dba77c19c9ed6b3873/mmcls/models/losses/focal_loss.py#L108) and the [doc](https://github.com/open-mmlab/mmclassification/blob/6beac5060f0c1d57141642dba77c19c9ed6b3873/mmcls/models/losses/cross_entropy_loss.py#L144)
> 
> The `loss` field only supports `dict` data, refer to this [line](https://github.com/open-mmlab/mmclassification/blob/6beac5060f0c1d57141642dba77c19c9ed6b3873/mmcls/models/heads/cls_head.py#L32)

I have checked the loss code existed under the 'losses' directory ( mmclassification/mmcls/models/losses ),so do I need to write the new loss code like demo below?

```
from typing import Tuple

import torch
from torch import nn, Tensor

from ..builder import LOSSES
from .utils import weight_reduce_loss

def loss_function(pred, target, weight, reduction='mean',
                avg_factor=None):

    ……
    <some code performing process loss>
    ……

    if weight is not None:
        assert weight.dim() == 1
        weight = weight.float()
        if pred.dim() > 1:
            weight = weight.reshape(-1, 1)
    loss = weight_reduce_loss(loss, weight, reduction, avg_factor)

    return loss


@LOSSES.register_module()
class LossName(nn.Module):
    def __init__(self, reduction='mean', loss_weight=1.0) -> None:
        super(LossName, self).__init__()
        self.m = m
        self.reduction = reduction
        self.loss_weight = loss_weight

    def forward(self,
                pred,
                target,
                weight=None,
                avg_factor=None,
                reduction_override=None):
        assert reduction_override in (None, 'none', 'mean', 'sum')
        reduction = (
            reduction_override if reduction_override else self.reduction)

        loss = loss_function(pred, target, weight, reduction=reduction,
                           avg_factor=avg_factor)

        return loss


if __name__ == "__main__":
    pass

```

am I right?

**Comment by Ezra-Yu** [comment]
You can achieve what you want with the above, but I think that is not universal, I prefer below:


Since you want to use multiple losses, this class MultiLoss needs to pass multiple loss configuration files.

```
from typing import Tuple

import torch
from torch import nn, Tensor

from ..builder import LOSSES, build_loss
from .utils import weight_reduce_loss


@LOSSES.register_module()
class MultiLoss(nn.Module):
    def __init__(self, loss_configs, reduction='mean') -> None:
        super(LossName, self).__init__()
        assert isinstance( loss_configs, list)
        self.losses = [build_loss(loss_cfg) for loss_cfg in loss_configs]
        self.reduction = reduction

    def forward(self,
                pred,
                target,
                weight=None,
                avg_factor=None,
                reduction_override=None):
        assert reduction_override in (None, 'none', 'mean', 'sum')
        reduction = (
            reduction_override if reduction_override else self.reduction)

        losses = [loss_function(pred, target, weight, reduction=reduction, avg_factor=avg_factor)
                       for loss_function in self.losses]
        
        return sum(losses)


if __name__ == "__main__":
    pass

```

**Comment by bmxbmx3** [comment]
It worked！Thanks.
And I have another question, will this multi loss feature be in the future development plan?

**Comment by Ezra-Yu** [comment]
We currently have no such plans, you can create a PR to add this feature, refer to [this link](https://mmclassification.readthedocs.io/en/latest/community/CONTRIBUTING.html).

Is the multi-loss useful in your project? It is very rare to use multi-loss in classification task.
